compiler/erlang_check.erl

```diff
@@ -245,29 +244,26 @@ check_module(File) ->
             report,
             % By adding debug_info, we ensure that the output of xref:m will
             % contain the caller MFAs too.
-            debug_info,
-            {i, AbsDir ++ "/include"},
-            {i, AbsDir ++ "/../include"},
-            {i, AbsDir ++ "/../../include"},
-            {i, AbsDir ++ "/../../../include"}],
-
-    case process_rebar_configs(AbsDir) of
-        {ok, RebarOpts} ->
-            code:add_patha(absname(AbsDir, "ebin")),
-            {CompileOpts0, OutDir} =
-                case get(outdir) of
-                    undefined ->
-                        % strong_validation = we only want validation, don't
-                        % generate beam file
-                        {[strong_validation], undefined};
-                    OutDir0 ->
-                        AbsOutDir = filename:join(AbsDir, OutDir0),
-                        {[{outdir, AbsOutDir}], AbsOutDir}
-                end,
-            CompileOpts = CompileOpts0 ++ Defs ++ RebarOpts,
+            debug_info],
+
+    BuildSystemOpts = case guess_build_system(AbsDir) of
+                          {rebar, RebarFiles} ->
+                              load_rebar_files(RebarFiles);
+                          {make, Makefiles} ->
+                              load_makefiles(Makefiles)
+                      end,
+    {ExtraOpts, OutDir} = case get(outdir) of
+                              undefined ->
+                                  {[strong_validation], undefined};
+                              OutDir0 ->
+                                  AbsOutDir = filename:join(AbsDir, OutDir0),
+                                  {[{outdir, AbsOutDir}], AbsOutDir}
+                          end,
+    case BuildSystemOpts of
+        {ok, Opts} ->
+            CompileOpts = Defs ++ Opts ++ ExtraOpts,
             log("Code paths: ~p~n", [code:get_path()]),
-            log("Compiling: compile:file(~p,~n    ~p)~n",
-                [AbsFile, CompileOpts]),
+            log("Compiling: compile:file(~p,~n	~p)~n", [AbsFile, CompileOpts]),
```

Comment:
(Bikeshed alert) Let's not put tab characters in the source code.

Just one example for the surprises it can cause: when I copy-pasted this line from Vim into the Erlang shell (I tried R16B03):

    "Compiling: compile:file(~p,~n	~p)~n"

I saw this:

    Eshell V5.10.4  (abort with ^G)
    1> "Compiling: compile:file(~p,~net_kernel:~p)~n"

Because "tab" was interpreted as a request for auto completion... Also, if someone reads the code, they will have no idea that that is a tab character and not a few spaces.

I suggest keeping the original "four spaces" version of this code. (So that if someone wants to do something with the output, they will be spared of their surprises caused by having tab in their input.)

Or if we have a good reason for changing the original log line to print a tab character, then I would prefer writing `"compile:file(~p,~n\t~p)~n"`.